cips/cip-6-second-blocks.md

```diff
@@ -0,0 +1,72 @@
+| cip | TBD |
+| - | - |
+| title | Reduce block time to six seconds |
+| description | Proposal to reduce block time on celestia-app to 6 seconds, from 12. |
+| author | Josh Stein ([@jcstein](https://github.com/jcstein)), Rootul Patel ([@rootulp](https://github.com/rootulp))|
+| discussions-to | <https://forum.celestia.org/t/cip-decrease-block-time-to-6-seconds/1836> |
+| status | Draft |
+| type | Standards Track |
+| category | Core |
+| created | 2024-10-09 |
+
+## Abstract
+
+This CIP proposes to reduce the block time on celestia-app to 6 seconds, from 12. This will double the throughput and reduce the time it takes for transactions to be finalized in half. This CIP also proposes to increase the `ttl-num-blocks` parameter in the mempool configuration from 5 to 12 to maintain consistency with the new block time.
+
+## Motivation
+
+The motivation for this CIP stems from a discussion in Core Devs Call 17, where it was proposed to reduce the block time to 6 seconds from 12 seconds.
+
+## Specification
+
+1. The block time in celestia-app SHOULD be reduced from 12 seconds to 6 seconds. Concretely, this implies decreasing `TimeoutCommit` to 1 seconds and `TimeoutPropose` to 3 seconds.
+
+1. All implementations of celestia-app SHALL adjust their block production mechanisms to conform to this new 6-second block time.
+
+1. The change in block time MUST be implemented at a specific block height, which SHALL be determined and agreed upon by the Celestia community via on-chain signaling by validators.
+
+1. Celestia consensus nodes SHOULD update their software to accommodate this change prior to the agreed-upon block height.
+
+1. Client applications interacting with the Celestia network SHOULD be updated to account for the faster block time, particularly in areas related to transaction confirmation and block finality.
+
+1. The implementation of this change SHOULD include appropriate testing and monitoring to ensure network stability during and after the transition.
+
+1. Documentation and APIs related to block time and block production MUST be updated to reflect this change.
+
+1. The `ttl-num-blocks` parameter in the mempool configuration SHALL be increased from 5 to 12. This change is necessary to maintain consistency with the new block time and ensure that transactions remain in the mempool for a similar duration as before.
+    1. Current default: `ttl-num-blocks = 5`
+    1. New default: `ttl-num-blocks = 12`
+
+1. This change SHALL be implemented alongside the block time reduction to ensure consistent behavior of the mempool across the network upgrade.
```

Comment:
[no change needed] just a note for other reviewers that we increased the default from `5` => `12` but that change doesn't occur alongside the block time reduction. The change occurs when people upgrade to v3.0.0 and regenerate their config files.

The block time reduction will happen one week later when the v2 -> v3 activation height occurs.